falcon/api.py

```diff
@@ -220,21 +220,24 @@ def __call__(self, env, start_response):  # noqa: C901
                 if self._independent_middleware:
                     for process_request in mw_req_stack:
                         process_request(req, resp)
+                        if resp.complete:
+                            break
                 else:
                     for process_request, process_response in mw_req_stack:
-                        if process_request:
+                        if process_request and not resp.complete:
```

Comment:
Unless I missed something I'm getting an impression that this combination has no test coverage:
* `independent_middleware=False`
* skipping `process_request` due to `resp.complete`

Reply:
Yikes, good catch! Covered now.